src/controllers/commands/CommandController.cpp

```diff
@@ -31,6 +32,136 @@
+            if (it != accemotes.emotes.end())
+            {
+                b.emplace<EmoteElement>(it->second,
+                                        MessageElementFlag::TwitchEmote);
+                continue;
+            }
+        }  // twitch emote
+
+        {  // bttv/ffz emote
+            if ((emote = bttvemotes.emote({words[i]})))
+            {
+                flags = MessageElementFlag::BttvEmote;
+            }
+            else if ((emote = ffzemotes.emote({words[i]})))
+            {
+                flags = MessageElementFlag::FfzEmote;
+            }
+            if (emote)
+            {
+                b.emplace<EmoteElement>(emote.get(), flags);
+                continue;
+            }
+        }  // bttv/ffz emote
+        {  // emoji/text
+            for (auto &variant : app->emotes->emojis.parse(words[i]))
+            {
+                constexpr const static struct {
+                    void operator()(EmotePtr emote, MessageBuilder &b) const
+                    {
+                        b.emplace<EmoteElement>(emote,
+                                                MessageElementFlag::EmojiAll);
+                    }
+                    void operator()(const QString &string,
+                                    MessageBuilder &b) const
+                    {
+                        auto linkString = b.matchLink(string);
+                        if (linkString.isEmpty())
+                        {
+                            b.emplace<TextElement>(string,
+                                                   MessageElementFlag::Text);
+                        }
+                        else
+                        {
+                            b.addLink(string, linkString);
+                        }
+                    }
+                } visitor;
+                boost::apply_visitor([&b](auto &&arg) { visitor(arg, b); },
+                                     variant);
+            }  // emoji/text
+        }
+    }
+
+    b->flags.set(MessageFlag::DoNotTriggerNotification);
+    b->flags.set(MessageFlag::Whisper);
+    auto messagexD = b.release();
+
+    app->twitch.server->whispersChannel->addMessage(messagexD);
+
+    auto overrideFlags = boost::optional<MessageFlags>(messagexD->flags);
+    overrideFlags->set(MessageFlag::DoNotLog);
+
+    if (getSettings()->inlineWhispers)
+    {
+        app->twitch.server->forEachChannel(
+            [&messagexD, overrideFlags](ChannelPtr _channel) {
+                _channel->addMessage(messagexD, overrideFlags);
+            });
+    }
+
+    return true;
+}
+
+bool appendWhisperMessageLocally(const QString &textNoEmoji)
+{
+    QString text = getApp()->emotes->emojis.replaceShortCodes(textNoEmoji);
+    QStringList words = text.split(' ', QString::SkipEmptyParts);
+
+    if (words.length() == 0)
+    {
+        return false;
+    }
+
+    QString commandName = words[0];
+
+    if (commandName == "/w")

```

Comment:
Make a `QStringList` containing "/w" and ".w" and use its contains operator to check if it's a whisper instead. make sure to supply the `Qt::CaseInsensitive` argument. This comment applies to other places you've touched that checks if something is a whisper

Reply:
https://github.com/fourtf/chatterino2/blob/b7d04a648ff7ab584c07f5d9d58defe89b393253/src/controllers/commands/CommandController.cpp#L158

https://github.com/fourtf/chatterino2/blob/b7d04a648ff7ab584c07f5d9d58defe89b393253/src/controllers/commands/CommandController.cpp#L260